tfutils/base.py

```diff
@@ -365,7 +363,7 @@ def run(sess,
         assert (step > old_step), (step, old_step)
         if train_results['loss'] > thres_loss:
             raise HiLossError('Loss {:.2f} exceeded the threshold {:.2f}'.format(train_results['loss'], thres_loss))
-        if step % saver.save_valid_freq == 0 and valid_targets is not None:
```

Comment:
Yeah -- we could write a commit hook and connect that to a test suite.   Maybe it's time to do that now.